File: ELIMINACION INICIO,ESP,FINAL/eliminacion.cs
using System;
using System.Collections.Generic;
using System.Globalization;

class Program
{
    static void imprimirlista(List<int>arr)
    {
        Console.WriteLine("[" + string.Join(", ", arr) + "]");

    }
    static void Main()
    {
        List<int>originalarr=new List<int> { 11,21,31,41,51,61};
        Console.WriteLine("eliminacion al inicio");
        List<int> inputarr=new List<int>(originalarr);
        int position= 0;

        Console.Write("antes: ");
        imprimirlista(inputarr);

        for (int k= position; k < inputarr.Count -1;k++)
        {
            inputarr[k]= inputarr[k+1];
        }
        inputarr.RemoveAt(inputarr.Count -1);

        Console.Write("despues: ");
        imprimirlista(inputarr);
        Console.WriteLine();



        Console.WriteLine("eliminacion especifico: ");
        inputarr= new List<int>(originalarr);
        position=3;

        Console.Write("antes: ");
        imprimirlista(inputarr);


        for (int k=position; k<inputarr.Count-1; k++)
        {
           inputarr[k]=inputarr[k+1];
        }
            inputarr.RemoveAt(inputarr.Count-1);
            
            Console.WriteLine("despues: ");
            imprimirlista(inputarr);
            Console.WriteLine();
                    
             
             Console.WriteLine("eliminacion final: ");
             inputarr=new List<int>(originalarr);

             Console.Write("antes: ");
             imprimirlista(inputarr);
             inputarr.RemoveAt(inputarr.Count-1);
             Console.Write("despues :");
             imprimirlista(inputarr);
    }
}

File: ELIMINACION INICIO,ESP,FINAL/eliminacion.cpp
#include <iostream>
#include <vector>

void imprimirvector(const std::vector<int>& arr) {
    std::cout<<"[";
    for (size_t u=0; u<arr.size();++u){
        std::cout<<arr[u];
        if(u<arr.size()-1) std::cout<<", ";
    }
    std::cout<<"]"<<std::endl;
}

int main() {
    std::vector<int> originalarr = {11,21,31,41,51,61};

    //eliminacion al inicio
    std::cout<<"eliminacion al inicio"<<std::endl;
    std::vector<int>inputarr=originalarr;
    int posicion=0;
    std::cout <<("antes: ");
    imprimirvector(inputarr);

    for(size_t i=posicion; i<inputarr.size()-1; ++i){
        inputarr[i]=inputarr[i+1];
    }
    inputarr.pop_back();
    std::cout <<("despues: ");
    imprimirvector(inputarr);
    std::cout<<std::endl;


    //eliminacion especifico
    std::cout<<"eliminacion especifico"<<std::endl;
    inputarr=originalarr;
    posicion=2; 
    std::cout <<("antes: ");
    imprimirvector(inputarr);
    for(size_t i=posicion; i<inputarr.size()-1; ++i){
        inputarr[i]=inputarr[i+1];
    }   
    inputarr.pop_back();
    std::cout <<("despues: ");
    imprimirvector(inputarr);
    std::cout<<std::endl;

    //eliminacion al final
    std::cout<<"eliminacion al final"<<std::endl;
    inputarr=originalarr;
    std::cout <<("antes: ");
    imprimirvector(inputarr);
    inputarr.pop_back();
    std::cout <<("despues: ");
    imprimirvector(inputarr);
    return 0;
}
